sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -175,6 +175,8 @@ private[sql] class HiveSessionCatalog(
     super.functionExists(name) || hiveFunctions.contains(name.funcName)
   }
 
+  override def externalFunctionExists(name: FunctionIdentifier): Boolean = functionExists(name)
```

Comment:
According to your logic, I think HiveSessionCatalog should override both `buildinFunctionExists` and `externalFunctionExists`. Like:

```
override def buildinFunctionExists(name: FunctionIdentifier): Boolean = {
   super.buildinFunctionExists(name) || hiveFunctions.contains(name.funcName)
}
override def externalFunctionExists(name: FunctionIdentifier): Boolean = functionExists(name) = {
  super.externalFunctionExists(name)
}
```

Reply:
@WeichenXu123 thanks very much for reviewing. Yes, I will add both in the HiveSessionCatalog.

Reply:
@WeichenXu123 thanks very much for reviewing. I am a little confused. So HiveSessionCatalog's builtinFunctionExists is essentially same as its parent. That is the reason i didn't override it in HiveSessionCatalog. However the logic to lookup an external function is different in HiveSessionCatalog as we also have to handle the special function "histogram_numeric". Thats why i choose to override the externalFunctionExists. One clarification is that builtinFunctionExists solely looks at FunctionRegistry to lookup a function.

1. builtInFunctionExists => Looks up a function in FunctionRegistry (same for both SessionCatalog and HiveSessionCatalog
2. externalFunctionExists => Looks up an external catalog to find the function. HiveSessionCatalog has one additional semantics to handle the special function called histogram_numeric.
3. as you point out, I need to change the override externalFunctionExists to this:
`override def externalFunctionExists(name: FunctionIdentifier): Boolean = {
    super.externalFunctionExists(name) || hiveFunctions.contains(name.funcName)`

Reply:
Oh, you mean functions like "histogram_numeric" should be regarded as externalFunction in hiveContext ?  I am not sure about this. But if that's right your current code is OK :)

Reply:
yes, seem "histogram_numeric" is not supported in spark natively yet, I think once this jira closed (https://issues.apache.org/jira/browse/SPARK-16280), we don't need these codes in the HiveSessionCatalog.